list.go

```diff
@@ -0,0 +1,242 @@
+// Copyright 2009 The Go Authors. All rights reserved.
```

Comment:
The `List` type probably shouldn't be part of the public exported API.  Maybe move this to an `internal` package?